skimage/filters/thresholding.py

```diff
@@ -236,16 +236,66 @@ def threshold_local(image, block_size, method='gaussian', offset=0,
     return thresh_image - offset
 
 
-def threshold_otsu(image, nbins=256):
+def _validate_image_histogram(image, hist, nbins=None):
+    """Ensure that either image or hist were given, return valid histogram.
+
+    If hist is given, image is ignored.
+
+    Parameters
+    ----------
+    image : array or None
+        Grayscale image.
+    hist : array, 2-tuple of array, or None
+        Histogram, either a 1D counts array, or an array of counts together
+        with an array of bin centers.
+    nbins : int, optional
+        The number of bins with which to compute the histogram, if `hist` is
+        None.
+
+    Returns
+    -------
+    counts : 1D array of float
+        The number of pixels falling in each intensity bin
+    bin_centers : 1D array
+        The value corresponding to the center of each intensity bin.
+
+    Raises
+    ------
+    ValueError : if image and hist are both None
+    """
+    if image is None and hist is None:
+        raise Exception("Either image or hist must be provided.")
+
+    if hist is not None:
+        if isinstance(hist, (tuple, list)):
+            counts, bin_centers = hist
+        else:
+            counts = hist
+            bin_centers = np.arange(counts.size)
+    else:
+        counts, bin_centers = histogram(
+                image.ravel(), nbins, source_range='image'
+            )
```

Comment:
Are the above two lines extra-indented, or is it just GitHub's rendering?

Reply:
this was my editor's indentation choice and I liked it! And it's PEP8 compliant. ;)

Reply:
(Did I already mention I hate black? =P)

Reply:
> this was my editor's indentation choice and I liked it! And it's PEP8 compliant. ;)

No, it's not! :confounded: See https://github.com/scikit-image/scikit-image/pull/5006#issuecomment-703691052:

* In the file skimage/filters/thresholding.py:

    Line 277:17: E126 continuation line over-indented for hanging indent
    Line 278:13: E121 continuation line under-indented for hanging indent

Reply:
> (Did I already mention I hate black? =P)

This I know! :rofl: I was surprised to see double quotes [here](https://github.com/scikit-image/scikit-image/pull/5006/files#diff-4f1fba6d7478009483dadb139f489db58130185867229af13354ae80a1b2ca3aR267). :wink:

Reply:
Maybe these double quote come from that Java developer who sent this PR ;-)

Reply:
Black can be told to leave quotes alone; that's my preference.